phoenix-core-client/src/main/java/org/apache/phoenix/jdbc/PhoenixStatement.java

```diff
@@ -1342,9 +1342,11 @@ public ExecutableShowTablesStatement(String schema, String pattern) {
         @Override
         public QueryPlan compilePlan(final PhoenixStatement stmt, Sequence.ValueOp seqAction)
             throws SQLException {
-            PreparedStatement delegateStmt = QueryUtil.getTablesStmt(stmt.getConnection(), null,
-                getTargetSchema(), getDbPattern(), null);
-            return ((PhoenixPreparedStatement) delegateStmt).compileQuery();
+            try (PreparedStatement delegateStmt =
```

Comment:
This is suspicious.
We keep using the query plan returned from the closed statement.

Reply:
thanks reviewing， rollbacked it.

Reply:
> This is suspicious. We keep using the query plan returned from the closed statement.

thanks reviewing， rollbacked it.